### DISCUSSION TITLE
how to use TerrainPositionInformation to get texture character is standing on?

### DISCUSSION THREAD

**Discussion by bardmask1** [spimort/TerraBrush#118] [Q&A] [answered]
i set a variable of `TerrainPositionInformation `from the characters world position then i get the `TerrainPositionTextureInformation`
from that, but it has all textures from my texture set? i assumed this was the correct way to get the texture because thats what the readme said to do.

**Comment by spimort** [comment, marked as answer]
That's correct!

You will receive a list of texture because at a given point, you could have multiple texture (example, a mix between grass and dirt).
The list will be sorted to received the most present texture first and the least texture last.

You will receive the information about the presence with the "Factor" property.

I hope this helps! 🙂

**Reply by bardmask1** [reply]
thanks, it turns out i was using it correctly except when putting in my position i was using the y vector instead of the z vector. its working now

**Reply by spimort** [reply]
Alright, cool! I'll close the discussion 🙂
